File: tools/repo_lint/tests/fixtures/python/all_docstring_violations.py
# Missing module docstring


# Missing function docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def no_doc():
    pass


# Missing parameter documentation
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def missing_params(x, y):
    """Function summary only."""
    return x + y


# Missing return documentation
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def missing_return():
    """Function summary only."""
    return 42


# Missing raises documentation
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def missing_raises():
    """Function summary only."""
    raise ValueError("error")


# Class without docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class NoDoc:
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def method(self):
        pass


# Class with missing method docstrings
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class PartialDocs:
    """Class has docstring."""

# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def no_doc_method(self):
        pass

# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def another_no_doc(self):
        return 1


# Incomplete docstring - missing sections
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def incomplete(x, y):
    """Just a summary line."""
    if x < 0:
        raise ValueError("negative")
    return x + y


# Wrong docstring format
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def wrong_format(a, b, c):
    """
    This uses single quotes instead of triple double quotes
    """
    return a + b + c


# Missing docstring for class method
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class MethodMissing:
    """Class docstring present."""

# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def __init__(self):
        pass

# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def compute(self, x):
        return x * 2


# Property without docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class PropertyMissing:
    """Class docstring present."""

    @property
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def value(self):
        return 42


# Static method without docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class StaticMissing:
    """Class docstring present."""

    @staticmethod
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def helper():
        return "help"


# Class method without docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
class ClassMethodMissing:
    """Class docstring present."""

    @classmethod
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def create(cls):
        return cls()


# Nested function without docstring
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def outer():
    """Outer function."""

# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    def inner():
        return 1

    return inner()


# Lambda (no docstring possible, but included for completeness)
calc = lambda x: x * 2


# Missing Examples section
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def no_examples():
    """Function summary.

    :param x: some param
    :returns: some value
    """
    pass


# Missing Notes section when complexity warrants it
# INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
def complex_algorithm(data):
    """Perform complex operation."""
    result = []
    for item in data:
        if item > 0:
            result.append(item * 2)
    return result

File: tools/repo_lint/tests/fixtures/rust/all-docstring-violations.rs
// Missing module documentation

// Function without doc comment
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn no_doc() {
    println!("no documentation");
}

// Function with incomplete doc
/// Just a summary
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn partial_doc(x: i32, y: i32) -> i32 {
    x + y
}

// Missing parameter documentation
/// Adds two numbers
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn missing_params(x: i32, y: i32) -> i32 {
    x + y
}

// Missing return documentation
/// Computes something
/// 
/// # Arguments
/// * `x` - first value
/// * `y` - second value
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn missing_return(x: i32, y: i32) -> i32 {
    x + y
}

// Missing examples
/// Complex function that does something
/// 
/// # Arguments
/// * `data` - input data
/// 
/// # Returns
/// Processed data
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn no_examples(data: Vec<i32>) -> Vec<i32> {
    data
}

// Struct without documentation
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
struct NoDocStruct {
    field1: i32,
    field2: String,
}

// Struct with incomplete documentation
/// A structure
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
struct PartialStruct {
    /// Documented field
    field1: i32,
    // Missing doc for this field
    field2: String,
}

// Enum without documentation
enum NoDocEnum {
    Variant1,
    Variant2,
}

// Trait without documentation
trait NoDocTrait {
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    fn method(&self) -> i32;
}

// Implementation without documentation
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
impl NoDocStruct {
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    fn new() -> Self {
        NoDocStruct {
            field1: 0,
            field2: String::new(),
        }
    }
    
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    fn method(&self) -> i32 {
        self.field1
    }
}

// Module without documentation
mod inner {
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
    pub fn helper() {}
}

// Const without documentation
const NO_DOC_CONST: i32 = 42;

// Static without documentation
static NO_DOC_STATIC: &str = "static";

// Type alias without documentation
type NoDocType = Vec<i32>;

// Missing panic documentation
/// Does something
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn may_panic(x: i32) -> i32 {
    if x < 0 {
        panic!("negative");
    }
    x
}

// Missing error documentation
/// Tries to do something
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn may_error(x: i32) -> Result<i32, String> {
    if x < 0 {
        Err("negative".to_string())
    } else {
        Ok(x)
    }
}

// Missing safety documentation
/// Unsafe function
unsafe fn unsafe_operation() {
    // Unsafe code
}

// Generic function without constraint documentation
/// Generic function
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
fn generic_func<T>(x: T) -> T {
    x
}

// Missing invariants documentation
/// Complex structure with invariants
// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
struct Complex {
    // Must be positive
    value: i32,
}

// INTENTIONAL VIOLATION - FOR TESTING ONLY - NOT FOR REVIEW
pub fn main() {}
